core/consensus/grandpa/impl/grandpa_impl.cpp

```diff
@@ -605,39 +608,38 @@ namespace kagome::consensus::grandpa {
 
   outcome::result<void> GrandpaImpl::applyJustification(
       const BlockInfo &block_info, const GrandpaJustification &justification) {
-    if (auto target_round = selectRound(justification.round_number)) {
-      return target_round->applyJustification(block_info, justification);
-    }
+    auto round = selectRound(justification.round_number);
+    if (round == nullptr) {
+      if (current_round_->bestPrevoteCandidate().number > block_info.number) {
+        return VotingRoundError::JUSTIFICATION_FOR_BLOCK_IN_PAST;
+      }
 
-    if (current_round_->bestPrevoteCandidate().number > block_info.number) {
-      return VotingRoundError::JUSTIFICATION_FOR_BLOCK_IN_PAST;
-    }
+      MovableRoundState round_state{
+          .round_number = justification.round_number,
+          .last_finalized_block = current_round_->lastFinalizedBlock(),
+          .votes = {},
+          .finalized = block_info};
 
-    MovableRoundState round_state{
-        .round_number = justification.round_number,
-        .last_finalized_block = current_round_->lastFinalizedBlock(),
-        .votes = {},
-        .finalized = block_info};
-
-    auto authorities_res =
-        authority_manager_->authorities(round_state.last_finalized_block, true);
-    if (not authorities_res.has_value()) {
-      BOOST_ASSERT(authorities_res.error().message().empty());
-    }
-    auto &authorities = authorities_res.value();
+      auto authorities_res = authority_manager_->authorities(
+          round_state.last_finalized_block, true);
+      if (not authorities_res.has_value()) {
+        BOOST_ASSERT(authorities_res.error().message().empty());
+      }
+      auto &authorities = authorities_res.value();
 
-    auto voters = std::make_shared<VoterSet>(authorities->id);
-    for (const auto &authority : *authorities) {
-      voters->insert(primitives::GrandpaSessionKey(authority.id.id),
-                     authority.weight);
-    }
+      auto voters = std::make_shared<VoterSet>(authorities->id);
+      for (const auto &authority : *authorities) {
+        voters->insert(primitives::GrandpaSessionKey(authority.id.id),
+                       authority.weight);
+      }
 
-    auto round = makeInitialRound(round_state, std::move(voters));
-    assert(round);
+      round = makeInitialRound(round_state, std::move(voters));
+      assert(round);
```

Comment:
We typically use BOOST_ASSERT so I guess it's better to stick to it for consistency.